LeagueOfPoro/CapsuleFarmerEvolved
# feat: Added Drop and 2FA Notifications and Toasts

## Summary of Changes

- Implemented NotificationManager Class to handle all notifications and toasts
- Added flags and their representive getters in Config.py for configurations as shown in [Configuration Wiki](https://github.com/LeagueOfPoro/CapsuleFarmerEvolved/wiki/Configuration)
- Added NotificationManager to FarmThread Class as parameter to alert on Fault and Sign in
- Added NotificationManager to Browser Class as parameter to alert on 2FA notification
- Added NotificationManager to Stats Class as parameter to alert on Drop notification
- Added default sound audio file and toast message icon
- Updated main.py to intialize NotificationManager
- Updated Pipfile with new libraries used (notify-py, wave, pyaudio)
- Updated gitignore for VS code stuff, sessions and logs

## Additional context
A reopening of PR #110 to implement a cross-platform sound notification and toast alerts for signing in, faults, 2FA or on recieving Drops. It has been implemented using Notify library that sends Toasts with option of adding sounds and multiple trial of other libraries such as beepy, PyAudio and SimpleAudio in hopes of finding one with no additional requirements for Mac OS and Linux.  A trial on any other library can be implemented easily by changing the functionality of playSoundOnly function in NotificationManager.py. It has been successfully tested on Windows 11.

## Testing instructions
- Try all combinations of all Flags in config.yaml regarding sound and notification [here](https://github.com/LeagueOfPoro/CapsuleFarmerEvolved/wiki/Configuration) and make sure it run successfully
- Test using custom audio files and adding soundPath to config.yaml

## How to download the PR for testing

### Using [GitHub CLI](https://cli.github.com/)
1. Clone this PR
2. Run `gh pr checkout 121` (Requires [GitHub CLI](https://cli.github.com/)) 
3. Follow the [Advanced Installation Guides from the Wiki](https://github.com/LeagueOfPoro/CapsuleFarmerEvolved/wiki)

### Using regular GIT
1. Fetch the PR `git fetch origin pull/121/head:master`
2. Checkout the branch `git checkout master`
3. Follow the [Advanced Installation Guides from the Wiki](https://github.com/LeagueOfPoro/CapsuleFarmerEvolved/wiki)
